webrender/src/renderer.rs

```diff
@@ -1390,14 +1396,25 @@ impl Renderer {
                                 &projection);
         }
 
+        let mut prev_blend_mode = BlendMode::None;
+
         for batch in &target.alpha_batcher.batches {
             let color_texture_id = batch.key.color_texture_id;
             let mask_texture_id = batch.key.mask_texture_id;
             let transform_kind = batch.key.flags.transform_kind();
             let has_complex_clip = batch.key.flags.needs_clipping();
-            let blending_enabled = batch.key.flags.needs_blending();
 
-            self.device.set_blend(blending_enabled);
+            if batch.key.blend_mode != prev_blend_mode {
+                match batch.key.blend_mode {
+                    BlendMode::None => {
+                        self.device.set_blend(false);
+                    }
+                    BlendMode::Alpha => {
+                        self.device.set_blend(true);
+                    }
+                }
```

Comment:
nit: could just be `self.device.set_blend(batch.key.blend_mode == BlendMode::Alpha)`